pkg/cloudprovider/providers/azure/azure.go

```diff
@@ -424,3 +441,87 @@ func initDiskControllers(az *Cloud) error {
 
 	return nil
 }
+
+// SetInformers sets informers for Azure cloud provider.
+func (az *Cloud) SetInformers(informerFactory informers.SharedInformerFactory) {
```

Comment:
if the goal is to get the zones used by the cluster for disks then we shouldn't need a watch for that. We can use the cached VM or we can even call `get nodes` every time we need to. Please remove the this watch

Reply:
This is used to get active zones from nodes. This is recommended way because

- We don't cache every VM in the cluster (remember we have removed list cache when adding that)
- Informers is a common pattern to get nodes information in kubernetes, it is used in many controllers and stable enough for our cases. Get nodes every time isn't a good idea, it should be cached.

Reply:
@khenidak The code logic is copied from GCE, I think we could use this in general. While I think we should add more logs in this func, node add/update/delete is not a frequent action, we could add more logs for diagnostics.

Reply:
I don't understand why the code is done elsewhere this way. Having it done this way somewhere else does not make it correct and/or the best way for us. And things might be different elsewhere. I am more concerned with this watch, in this context.

Watches are nice, but they are expensive. And there are way to many of them currently in the ecosystem. You are essentially establishing additional long running connection to api-server and caching data locally. For an object that does not change that frequently. However due to the node heart beat, this channel will be `needlessly` always busy with updates. My argument is. Having a `current` list of zones, does not require watch. a simple `list-node` will suffice with less code, less cpu cycle and easier to maintain.

Please change.

Reply:
Talked with KAL offline, we agreed to use informers to fetch node's information.